pages/data-providers/[data-provider-id]/doi.jsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react'
+
+import { withGlobalStore } from 'store'
+import Table from 'components/table'
+import NumericValue, { formatNumber } from 'components/numeric-value'
+import ExportButton from 'components/export-button'
+import EnrichmentChart from 'components/enrichment-chart'
+import Markdown from 'components/markdown'
+import * as texts from 'texts/doi'
+import { Card, Icon } from 'design'
+
+// TODO: Remove once cards are in @oacore/design
+// eslint-disable-next-line
+import styles from './doi.css'
+
+// const formatPercent = (number, precision = 2) =>
+//   formatNumber(number * 100, { precision })
+
+const formatDOI = entity => {
+  const { crossrefDoi, repoDoi: originDoi } = entity
+
+  if (crossrefDoi && !originDoi) {
+    return (
+      <>
+        {crossrefDoi}{' '}
+        <Icon
+          className={styles.newIcon}
+          src="/design/icons.svg#new"
+          alt="New"
+        />
+      </>
+    )
+  }
+
+  return originDoi
+}
+
+// const useLoading = (func, deps = []) =>
+//   useMemo(() => {
+//     const fetched = deps.reduce((done, value) => done && value != null, true)
+//     if (!fetched) return null
+//     return func()
+//   }, deps)
+
+const useDefault = (value, substitute = null) =>
+  value == null ||
+  value === Infinity ||
+  value === -Infinity ||
+  Number.isNaN(value)
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isFinite This function will cover +-Infinity and isNaN cases